src/overlays/actors/ovl_kaleido_scope/z_kaleido_item.c

```diff
@@ -1,9 +1,813 @@
+                    }
+                }
+            }
+
+            gSPVertex(POLY_OPA_DISP++, &pauseCtx->itemVtx[j + 0], 4, 0);
+            KaleidoScope_DrawQuadTextureRGBA32(globalCtx->state.gfxCtx, gItemIcons[gSaveContext.inventory.items[i]], 32,
+                                               32, 0);
+        }
+    }
+
+    if (pauseCtx->cursorSpecialPos == 0) {
+        KaleidoScope_DrawCursor(globalCtx, PAUSE_ITEM);
+    }
+
+    gDPPipeSync(POLY_OPA_DISP++);
+    gDPSetCombineLERP(POLY_OPA_DISP++, PRIMITIVE, ENVIRONMENT, TEXEL0, ENVIRONMENT, TEXEL0, 0, PRIMITIVE, 0, PRIMITIVE,
+                      ENVIRONMENT, TEXEL0, ENVIRONMENT, TEXEL0, 0, PRIMITIVE, 0);
+
+    for (i = 0; i < 15; i++) {
+        if ((gAmmoItems[i] != ITEM_NONE) && (gSaveContext.inventory.items[i] != ITEM_NONE)) {
+            KaleidoScope_DrawAmmoCount(pauseCtx, globalCtx->state.gfxCtx, gSaveContext.inventory.items[i]);
+        }
+    }
+
+    CLOSE_DISPS(globalCtx->state.gfxCtx, "../z_kaleido_item.c", 516);
+}
+
+static s16 sCButtonPosX[] = { 660, 900, 1140 };
+static s16 sCButtonPosY[] = { 1100, 920, 1100 };
+
+void KaleidoScope_UpdateItemEquip(GlobalContext* globalCtx) {
+    static s16 D_8082A488 = 0;
+    PauseContext* pauseCtx = &globalCtx->pauseCtx;
+    Vtx* bowItemVtx;
+    u16 offsetX;
+    u16 offsetY;
+
+    if (sEquipState == 0) {
+        pauseCtx->equipAnimAlpha += 14;
+        if (pauseCtx->equipAnimAlpha > 255) {
+            pauseCtx->equipAnimAlpha = 254;
+            sEquipState++;
+        }
+        sEquipAnimTimer = 5;
+        return;
+    }
+
+    if (sEquipState == 2) {
+        D_8082A488--;
+
+        if (D_8082A488 == 0) {
+            pauseCtx->equipTargetItem -= 0xBF - ITEM_BOW_ARROW_FIRE;
+            pauseCtx->equipTargetSlot = SLOT_BOW;
+            sEquipMoveTimer = 6;
+            WREG(90) = 320;
+            WREG(87) = WREG(91);
+            sEquipState++;
+            Audio_PlaySoundGeneral(NA_SE_SY_SYNTH_MAGIC_ARROW, &D_801333D4, 4, &D_801333E0, &D_801333E0, &D_801333E8);
+        }
+        return;
+    }
+
+    if (sEquipState == 1) {
+        bowItemVtx = &pauseCtx->itemVtx[12];
+        offsetX = ABS(pauseCtx->equipAnimX - bowItemVtx->v.ob[0] * 10) / sEquipMoveTimer;
+        offsetY = ABS(pauseCtx->equipAnimY - bowItemVtx->v.ob[1] * 10) / sEquipMoveTimer;
+    } else {
+        offsetX = ABS(pauseCtx->equipAnimX - sCButtonPosX[pauseCtx->equipTargetCBtn]) / sEquipMoveTimer;
+        offsetY = ABS(pauseCtx->equipAnimY - sCButtonPosY[pauseCtx->equipTargetCBtn]) / sEquipMoveTimer;
+    }
+
+    if (pauseCtx->equipTargetItem >= 0xBF) {
+        if (pauseCtx->equipAnimAlpha < 254) {
+            pauseCtx->equipAnimAlpha += 14;
+            if (pauseCtx->equipAnimAlpha > 255) {
+                pauseCtx->equipAnimAlpha = 254;
+            }
+            sEquipAnimTimer = 5;
+            return;
+        }
+    }
+
+    if (sEquipAnimTimer == 0) {
+        WREG(90) -= WREG(87) / sEquipMoveTimer;
```

Comment:
I think the github suggestion is crap for large changes. Basically merge both conditions at 531 and 532, change the `if` to an `else if` at 542 to remove the early `return`. 

```suggestion
    if (pauseCtx->equipTargetItem >= 0xBF && pauseCtx->equipAnimAlpha < 254) {
            pauseCtx->equipAnimAlpha += 14;
            if (pauseCtx->equipAnimAlpha > 255) {
                pauseCtx->equipAnimAlpha = 254;
            }
            sEquipAnimTimer = 5;
    }
    else if (sEquipAnimTimer == 0) {
        WREG(90) -= WREG(87) / sEquipMoveTimer;
```

Reply:
I'll merge the conditions at 531 and 532, but imo the early return is better since it's a special case and the general control flow of this function uses other early returns that can't really be avoided